projects/components/src/snackbar/snackbar.component.ts

```diff
@@ -0,0 +1,67 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { MAT_SNACK_BAR_DATA } from '@angular/material/snack-bar';
+import { IconType } from '@hypertrace/assets-library';
+import { assertUnreachable } from '@hypertrace/common';
+import { Observer } from 'rxjs';
+import { IconSize } from '../icon/icon-size';
+
+@Component({
+  selector: 'ht-snackbar',
+  styleUrls: ['./snackbar.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="snackbar-container">
+      <ht-icon
+        [ngClass]="this.matData.mode"
+        class="status-icon"
+        [icon]="this.iconType"
+        size="${IconSize.Small}"
+      ></ht-icon>
+      <div class="text">{{ this.matData.message }}</div>
+      <ht-icon
+        class="dismiss-icon"
+        icon="${IconType.CloseCircleFilled}"
+        size="${IconSize.Small}"
+        (click)="this.close()"
+      ></ht-icon>
+    </div>
+  `
+})
+export class SnackbarComponent {
+  public iconType?: IconType;
+  public constructor(@Inject(MAT_SNACK_BAR_DATA) public readonly matData: SnackbarData) {
+    this.setIconType(this.matData.mode);
+  }
+
+  public setIconType(mode: SnackbarMode): void {
+    switch (mode) {
```

Comment:
nit: functions that mutate instance variables are generally more brittle. This is a tiny nit in this case, but a good practice more generally. If this one returned an icon type, it'd be pure function, and the less functions that do mutation, the easier it is to understand the logic flows. Specifically in this cases, with a switch statement, allows avoiding `break` which is always good and it would allow iconType to be readonly because the assignment is all in the constructor.

Also, doesn't need to be public